### DISCUSSION TITLE
jsPsychSurvey dropdown: width issue?

### DISCUSSION THREAD

**Discussion by ChristopheBossens** [jspsych/jsPsych#3285] [General]
Hi,

I am in the process of converting some of our surveys to the updated jsPsychSurvey plugin but I ran into an issue using the dropdown survey component. For some reason I cannot set the width properly so it shows way to small:
![screenshot](https://github.com/jspsych/jsPsych/assets/7238485/a2015708-a8fb-44e2-afef-7c899d3aed74)

I am including the following resources:
```javascript
<script src="https://unpkg.com/@jspsych/plugin-survey@1.0.0"></script>
<link rel="stylesheet" href="https://unpkg.com/@jspsych/plugin-survey@1.0.0/css/survey.css">
```

and using the following code to display the dropdown:
```javascript
let dropdown = {
        type: jsPsychSurvey,
        survey_json: {
            showQuestionNumbers: false,
            "elements": [
                {
                    "type": "dropdown",
                    "name": "car",
                    "title": "Which is the brand of your car?",
                    "isRequired": true,
                    "showNoneItem": true,
                    "showOtherItem": true,
                    "choices": [ "Ford", "Vauxhall", "Volkswagen", "Nissan", "Audi", "Mercedes-Benz", "BMW", "Peugeot", "Toyota", "Citroen" ]
                }
            ],
            "showQuestionNumbers": false,
            "fitToContainer": true
        }
    }
```

I have been playing with the min and minWidth properties of the dropdown element but they do not seem to have any effect. Any ideas on what I am doing wrong here?

**Comment by Shaobin-Jiang** [comment]
Use this in your css:

```css
.jspsych-question-content {
    min-width: 50% !important;
}
```

**Comment by becky-gilbert** [comment]
Yikes, thanks for flagging this @ChristopheBossens! I'll include a fix in the next release.

**Reply by becky-gilbert** [reply]
In the meantime, you can fix this by adding `questionDescriptionLocation: "underInput"` to the survey_json:
```javascript
let dropdown = {
  type: jsPsychSurvey,
  survey_json: {
    showQuestionNumbers: false,
    questionDescriptionLocation: "underInput",
      "elements": [
      // ... etc ...
```

**Reply by Shaobin-Jiang** [reply]
I tried this and it does not work for me.

**Reply by becky-gilbert** [reply]
Apologies, you're absolutely right @Shaobin-Jiang. (The reason the "questionDescriptionLocation" change worked for me was because I had a long `description` in my example, and the description text was extending the width of the input element when its location was "underInput".)

I agree with your solution of adjusting the `width` of the `jspsych-question-content` class. I used `width: 100%`, which gives me this:

![Screenshot 2024-05-07 at 2 06 48 PM](https://github.com/jspsych/jsPsych/assets/9041788/d4f393c3-27b3-4458-9adf-6c385212c62b)

Sorry again about this - I'll fix it ASAP!

**Comment by becky-gilbert** [comment]
@ChristopheBossens @Shaobin-Jiang I'm putting together a fix for this here: https://github.com/jspsych/jsPsych/pull/3287 which adds `width: 100%` to the `jspsych-question-content` class. If either of you are unhappy with this solution can you please let me know? Thanks!

**Reply by Shaobin-Jiang** [reply]
LGTM.

**Comment by becky-gilbert** [comment]
Hi @ChristopheBossens, this problem should now be fixed in the latest survey-plugin release: https://github.com/jspsych/jsPsych/releases/tag/%40jspsych%2Fplugin-survey%401.0.1 Please let me know if this works for you!
